File: 백준/Gold/1806. 부분합/부분합.py
import sys
# sys.stdin = open("input.txt", "r")

N, target = map(int, input().split())
numArr = list(map(int, input().split()))

def solve():
    left, right = 0, 0
    sum = 0
    minLen = sys.maxsize
    
    while(True):
        if sum >= target:
            minLen = min(minLen, right - left)
            sum = sum - numArr[left]
            left = left + 1
        
        elif right == N:
            break
        
        elif sum < target:
            sum = sum + numArr[right]
            right = right + 1

    if minLen == sys.maxsize:
        print(0)
    else:
        print(minLen)
solve()

File: 백준/Gold/1806. 부분합/부분합.cc
#include <iostream>
#include <vector>
#include <algorithm>
#include <climits>

using namespace std;

int N, target;
vector<int> numArr;

void input() {
    cin >> N >> target;
    numArr.resize(N);
    for (int i = 0; i < N; ++i) {
        cin >> numArr[i];
    }
}

void solution() {
    int left = 0, right = 0, sum = 0, minLen = INT_MAX;

    while (true) {
        if (sum >= target) {
            minLen = min(minLen, right - left);
            sum -= numArr[left];
            ++left;
        } else if (right == N) {
            break;
        } else if (sum < target) {
            sum += numArr[right];
            ++right;
        }
    }

    if (minLen == INT_MAX) {
        cout << 0 << endl;
    } else {
        cout << minLen << endl;
    }
}

void solve() {
    input();
    solution();
}

int main() {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    // freopen("input.txt", "r", stdin);
    solve();
    return 0;
}